src/main/java/com/example/MicroblogSpringController.java

```diff
@@ -0,0 +1,40 @@
+package com.example;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.util.List;
+
+/**
+ * Created by Ben on 1/18/17.
+ */
+@Controller
+public class MicroblogSpringController {
+    @Autowired
+    MessageRepository messages;
+
+    @RequestMapping(path = "/", method = RequestMethod.GET)
+    public String home(Model model) {
+        List<Message> messageList = (List) messages.findAll();
+        model.addAttribute("messages", messageList);
+        return "home";
+    }
+
+    @RequestMapping(path = "/add-message", method = RequestMethod.POST)
+    public String addMessage(String message) {
+        Message message1 = new Message(message);
+        messages.save(message1);
+        return "redirect:/";
+    }
+
+    @RequestMapping(path = "/edit-message", method = RequestMethod.POST)
+    public String editMessage(String message, int messageId) {
+        Message current = messages.findOne(messageId);
+
+        current.setMessage(message);
+        return "redirect:/";
```

Comment:
This is certainly the code to do this, but we don't have a template that calls this code!